File: app/lib/src/config/supabase_startup.dart
import 'supabase_public_config.dart';

enum SupabaseStartupStatus { missingConfiguration, ready, failed }

typedef SupabaseStarter =
    Future<void> Function(String url, String publishableKey);

/// Initializes Supabase only when both public values are present.
Future<SupabaseStartupStatus> prepareSupabase(
  SupabasePublicConfig config,
  SupabaseStarter start,
) async {
  if (!config.isComplete) {
    return SupabaseStartupStatus.missingConfiguration;
  }
  try {
    await start(config.trimmedUrl, config.trimmedPublishableKey);
    return SupabaseStartupStatus.ready;
  } on Object {
    return SupabaseStartupStatus.failed;
  }
}

File: admin/src/config/supabaseStartup.ts
import {
  isSupabaseConfigComplete,
  trimmedPublishableKey,
  trimmedUrl,
  type SupabasePublicConfig,
} from './supabasePublicConfig.ts'

export type SupabaseStartupStatus = 'missingConfiguration' | 'ready' | 'failed'

export type SupabaseStarter = (
  url: string,
  publishableKey: string,
) => Promise<void>

/** Initializes Supabase only when both public values are present. */
export async function prepareSupabase(
  config: SupabasePublicConfig,
  start: SupabaseStarter,
): Promise<SupabaseStartupStatus> {
  if (!isSupabaseConfigComplete(config)) {
    return 'missingConfiguration'
  }

  try {
    await start(trimmedUrl(config), trimmedPublishableKey(config))
    return 'ready'
  } catch {
    return 'failed'
  }
}
